src/context.cc

```diff
@@ -23,6 +23,25 @@
 #include "include/proxy-wasm/context.h"
 #include "include/proxy-wasm/wasm.h"
 
+#define CHECK_FAIL(_call, _stream_type, _return_open, _return_closed)                              \
```

Comment:
This should probably be a real function.

Reply:
It conditionally returns sometimes.  That isn't something that can be done with a function.